src/metrics.cpp

```diff
@@ -192,16 +192,22 @@ void ThreadShowMetricsScreen()
     while (true) {
         // Number of lines that are always displayed
         int lines = 1;
+        int cols = 80;
 
         // Get current window size
-        struct winsize w;
-        ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
+        if (isatty(STDOUT_FILENO)) {
+          struct winsize w;
+          ioctl(STDOUT_FILENO, TIOCGWINSZ, &w);
```

Comment:
``ioctl`` [can fail](http://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html), in which case ``w`` will be uninitialized:
> Upon successful completion, ioctl() shall return a value other than -1 that depends upon the STREAMS device control function. Otherwise, it shall return -1 and set errno to indicate the error.

(I know this was wrong before your patch.)

So this should be something like:
```
if (isatty(STDOUT_FILENO)) {
    struct winsize w;
    w.ws_col = 0;
    if (ioctl(STDOUT_FILENO, TIOCGWINSZ, &w) != -1 && w.ws_col != 0) {
        cols = w.ws_col;
    }
}
```